tests/core/pyspec/eth2spec/test/electra/epoch_processing/test_process_pending_consolidations.py

```diff
@@ -185,3 +191,158 @@ def test_all_consolidation_cases_together(spec, state):
         assert state.balances[target_index[i]] == pre_balances[target_index[i]]
     # First consolidation is processed, second is skipped, last two are left in the queue
     state.pending_consolidations = pre_pending_consolidations[2:]
+
+
+@with_electra_and_later
+@spec_state_test
+def test_pending_consolidation_future_epoch(spec, state):
+    current_epoch = spec.get_current_epoch(state)
+    source_index = spec.get_active_validator_indices(state, current_epoch)[0]
+    target_index = spec.get_active_validator_indices(state, current_epoch)[1]
+    # initiate source exit
+    spec.initiate_validator_exit(state, source_index)
+    # set withdrawable_epoch to exit_epoch + 1
+    state.validators[source_index].withdrawable_epoch = state.validators[source_index].exit_epoch + spec.Epoch(1)
+    # append pending consolidation
+    state.pending_consolidations.append(
+        spec.PendingConsolidation(source_index=source_index, target_index=target_index)
+    )
+    # Set the target withdrawal credential to eth1
+    eth1_withdrawal_credential = (
+        spec.ETH1_ADDRESS_WITHDRAWAL_PREFIX + b"\x00" * 11 + b"\x11" * 20
+    )
+    state.validators[target_index].withdrawal_credentials = eth1_withdrawal_credential
+
+    # Advance to withdrawable_epoch - 1 with full participation
+    target_epoch = state.validators[source_index].withdrawable_epoch - spec.Epoch(1)
+    while spec.get_current_epoch(state) < target_epoch:
+        next_epoch_with_full_participation(spec, state)
+
+    # Obtain state before the call to process_pending_consolidations
+    state_before_consolidation = compute_state_by_epoch_processing_to(spec, state, "process_pending_consolidations")
+
+    yield from run_epoch_processing_with(spec, state, "process_pending_consolidations")
```

Comment:
Another solution is to make `run_epoch_processing_with` return values:
```python
def run_epoch_processing_with(spec, state, process_name: str):
    """
    Processes to the next epoch transition, up to and including the sub-transition named ``process_name``
      - pre-state ('pre'), state before calling ``process_name``
      - post-state ('post'), state after calling ``process_name``
    """
    run_epoch_processing_to(spec, state, process_name)
    pre = state.copy() 
    yield 'pre', pre
    getattr(spec, process_name)(state)
    yield 'post', state

    return pre, state
```

and call it with return values in tests:
```python
    pre, post = yield from run_epoch_processing_with(spec, state, "process_pending_consolidations")
```